src/material.js

```diff
@@ -326,16 +327,37 @@ class gltfMaterial extends GltfObject
                 if (this.sheenColorIntensityTexture !== undefined)
                 {
                     this.sheenColorIntensityTexture.samplerName = "u_sheenColorIntensitySampler";
-                    this.parseTextureInfoExtensions(this.sheenColorIntensityTexture, "SheenColorIntensityTexture");
+                    this.parseTextureInfoExtensions(this.sheenColorIntensityTexture, "SheenColorIntensity");
                     this.textures.push(this.sheenColorIntensityTexture);
-                    this.defines.push("HAS_SHEEN_COLOR_INTENSITY_TEXTURE_MAP 1");
+                    this.defines.push("HAS_SHEEN_COLOR_INTENSITY_MAP 1");
                     this.properties.set("u_sheenColorIntensityUVSet", this.sheenColorIntensityTexture.texCoord);
                 }
                 this.properties.set("u_SheenIntensityFactor", sheenFactor);
                 this.properties.set("u_SheenColorFactor", sheenColor);
                 this.properties.set("u_SheenRoughness", sheenRoughness);
             }
 
+            //KHR Extension Specular
+            // See https://github.com/ux3d/glTF/tree/KHR_materials_pbrClearcoat/extensions/2.0/Khronos/KHR_materials_specular
+            let specularFactor = 0.5;
+            if(this.extensions.KHR_materials_specular)
+            {
+                this.defines.push("MATERIAL_METALLICROUGHNESS_SPECULAROVERRIDE 1");
+
+                if(this.extensions.KHR_materials_specular.specularFactor !== undefined)
+                {
+                    specularFactor = this.extensions.KHR_materials_specular.specularFactor;
+                }
+                if (this.MetallicRoughnessSpecularTexture !== undefined)
+                {
+                    this.MetallicRoughnessSpecularTexture.samplerName = "u_MetallicRoughnessSpecularSampler";
```

Comment:
why is this called `MetallicRoughnessSpecularTexture` when the specification calls it `specularTexture`?

Reply:
there are already specular elements in the whole code, try to keep it together with the naming